pkg/wsman/amt/messagelog/log.go

```diff
@@ -104,16 +104,20 @@ func (messageLog Service) Pull(enumerationContext string) (response Response, er
 
 // GetRecords retrieves multiple records from event log.
 // The IterationIdentifier input parameter is a numeric value (starting at 1) which is the position of the first record in the log that should be extracted.
-// MaxReadRecords is set to 390.  If NoMoreRecords returns false, call this again setting the identifier to the start of the next IterationIdentifier.
-func (messageLog Service) GetRecords(identifier int) (response Response, err error) {
+// MaxReadRecords is set to 390 by default.  If NoMoreRecords returns false, call this again setting the identifier to the start of the next IterationIdentifier.
+func (messageLog Service) GetRecords(identifier, maxReadRecords int) (response Response, err error) {
 	if identifier < 1 {
 		identifier = 1
 	}
 
+	if maxReadRecords < 1 || maxReadRecords > 400 {
```

Comment:
is 400 the maximum the AMT accepts? 395 is acceptable? Seems odd for the 10 record allowance between 390 and 400. 400 should ideally reflect the maximum AMT accepts and probably have a const value defined for so its clear that its the maximum. Additionally, i think if the value of maxReadRecords exceeds the maximum, the assignment should be the maximum not the default. I think if its 0 -- thats when you would apply the default.

Reply:
1.The AMT documentation confirms that 400 is the maximum number of records that can be returned in a single GetRecords call. Reference: [AMT Implementation and Reference Guide](https://software.intel.com/sites/manageability/AMT_Implementation_and_Reference_Guide/default.htm?turl=HTMLDocuments%2FWS-Management_Class_Reference%2FAMT_MessageLog.htm)

2. While the documentation does not specify a minimum number below 400, we currently use 390 as our default value, aligning with MeshCommander. This has been proven to work reliably, although theoretically, we could use 395.

Reply:
To be more clear, this is my suggestion:

``` suggestion
	if maxReadRecords < 1 {
	maxReadRecords = DefaultPageSize
	} else if maxReadRecords > MaxPageSize {
	maxReadRecords = MaxPageSize
        }
```
where MaxPageSize = 400 (although im not seeing where this is specified in the docs)
and DefaultPageSize is more like 100